graphdb/janus/src/main/java/org/apache/atlas/repository/graphdb/janus/SearchContextCache.java

```diff
@@ -1,49 +1,73 @@
 
-import com.google.common.cache.Cache;
-import com.google.common.cache.CacheBuilder;
+    public static final String INVALID_SEQUENCE = "invalid_sequence";
 
-import java.util.HashMap;
-import java.util.Map;
-import java.util.concurrent.TimeUnit;
 
-public class SearchContextCache {
-    private static final Cache<String, HashMap<Integer, String>> searchContextCache = CacheBuilder.newBuilder()
-            .maximumSize(200)
-            .expireAfterWrite(30, TimeUnit.SECONDS)
-            .build();
+    public SearchContextCache(@Qualifier("redisServiceImpl") RedisService redisService) {
+        SearchContextCache.redisService = redisService;
+    }
+
 
     public static void put(String key, Integer sequence, String esAsyncId) {
-        HashMap<Integer, String> entry = new HashMap<>();
-        entry.put(sequence, esAsyncId);
-        searchContextCache.put(key, entry);
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("putInCache");
+       try {
+           // Build the string in format `sequence/esAsyncId` and store it in redis
+           String val = sequence + "/" + esAsyncId;
+           redisService.putValue(key, val);
+       } finally {
+           RequestContext.get().endMetricRecord(metric);
+       }
     }
-    public static HashMap<Integer, String> get(String key){
-        return searchContextCache.getIfPresent(key);
+    public static String get(String key) {
+        try {
+            return redisService.getValue(key);
+        } catch (Exception e) {
+            return null;
+        }
+
     }
 
     public static String getESAsyncSearchIdFromContextCache(String key, Integer sequence){
-        //Get the context cache for the given key
-        HashMap<Integer, String> contextCache = get(key);
-        if(contextCache == null || sequence == null){
-            return null;
-        }
-        //Find the highest sequence number
-        int maxStoredSequence = 0;
-        for (Integer seq : contextCache.keySet()) {
-            if (seq > maxStoredSequence) {
-                maxStoredSequence = seq;
+        AtlasPerfMetrics.MetricRecorder metric = RequestContext.get().startMetricRecord("getESAsyncSearchIdFromContextCache");
+        try {
+            //Get the context cache for the given key
+            String contextCache = get(key);
+            if(contextCache == null || sequence == null){
+                return null;
             }
+            // Split the context cache to get the sequence and ESAsyncId
+            String[] contextCacheSplit = contextCache.split("/");
+            if(contextCacheSplit.length != 2){
+                return null;
+            }
+            int seq = Integer.parseInt(contextCacheSplit[0]);
+            if(sequence > seq){
+                return contextCacheSplit[1];
+            } else if (sequence < seq) {
+                return INVALID_SEQUENCE;
+            }
+            return null;
+        } finally {
```

Comment:
No exceptions handled in this method as well. Can `Integer.parseInt` throw an exception?

Reply:
Handled in parent function `processRequestWithSameSearchContextId`